sycamore/transforms/embed.py

```diff
@@ -19,17 +19,23 @@
 from sycamore.utils import batched, generate_map_batch_function, generate_map_batch_class_from_callable
 
 
+def _pre_process_document(document: Document) -> str:
+    return document.text_representation if document.text_representation is not None else ""
```

Comment:
I would agree with what Henry said. Some of the embedding models will break on empty strings (which is why I did it differently for OpenAI/Bedrock). Maybe better to skip empty passages.

Reply:
Maybe could do something as simple as if you return null from the preprocess function then skip embedding the element.